src/coreclr/vm/callstubgenerator.cpp

```diff
@@ -2730,13 +2944,26 @@ void CallStubGenerator::ProcessArgument(ArgIterator *pArgIt, ArgLocDesc& argLocD
 #if LOG_COMPUTE_CALL_STUB
         printf("m_byteStackSize=%d\n", (int)argLocDesc.m_byteStackSize);
 #endif // LOG_COMPUTE_CALL_STUB
+#ifdef TARGET_ARM
+        if (splitRequired)
+        {
+            pRoutines[m_routineIndex++] = GetStackRoutine_4B();
+            pRoutines[m_routineIndex++] = argLocDesc.m_byteStackIndex;
+            pRoutines[m_routineIndex++] = argLocDesc.m_byteStackSize;
+        }
+        else
+#endif // TARGET_ARM
         if (m_s1 == NoRange) // No active range yet
         {
             // Start a new range
             m_s1 = argLocDesc.m_byteStackIndex;
             m_s2 = m_s1 + argLocDesc.m_byteStackSize - 1;
         }
-        else if ((argLocDesc.m_byteStackIndex == m_s2 + 1) && (argLocDesc.m_byteStackSize >= 8) && (!pArgIt || !pArgIt->IsArgPassedByRef()))
+#ifdef ENREGISTERED_PARAMTYPE_MAXSIZE
+        else if ((argLocDesc.m_byteStackIndex == m_s2 + 1) && (argLocDesc.m_byteStackSize >= TARGET_POINTER_SIZE) && (!pArgIt || !pArgIt->IsArgPassedByRef()))
```

Comment:
A nit - can you please keep only the extra part of the condition under the ifdef here?